cloud/cloud.go

```diff
@@ -13,6 +13,10 @@ import (
 // ProviderSettings exposes provider-specific configuration settings for a Manager.
 type ProviderSettings interface {
 	Validate() error
+
+	// If zone is specified, returns the provider settings for that region.
+	// This is currently only being implemented for EC2 hosts.
+	FromDistroSettings(distro.Distro, string) error
```

Comment:
Does this change functionality in anything other than EC2? If not, I think you can simplify this by making it a separate interface with just a single method. (Even if it changes some functionality in non-EC2, but mostly only changes functionality in EC2, it might still make sense to have a separate interface.) 

For example, we have a separate interface for Cloud managers that support batch operations (at the moment, only EC2):

```
// BatchManager is an interface for cloud providers that support batch operations.
type BatchManager interface {
	// GetInstanceStatuses gets the status of a slice of instances.
	GetInstanceStatuses(context.Context, []host.Host) ([]CloudStatus, error)
}
```

And then in the `cloudHostReadyJob` we check to see if the concrete `cloud.Manager` implements `BatchManager`:

```
if batch, ok := m.(cloud.BatchManager); ok {
```

With this pattern you would need only implement `FromDistroSettings` where it actually make sense.

Another reason to do this is that it seems like most providers implement the exact same `FromDistroSettings`, so instead of duplicating that code, you could have it in a shared place, so you'd do something like:

```
if r, ok := m.(cloud.RegionProvider); ok {
  r.FromDistroSettings(d, myRegion)
} else {
  // stuff that you currently have in every other distro's `FromDistroSettings`
}
```

Reply:
Discussed in person. Resolving.